### PR TITLE
[LPT] Dequantization constant output precision handling extending

### PR DESCRIPTION
**Pull request by eshoguli**
Issues: #42033 &  #42236

Accuracy validation: developer-services/job/accuracy/614/
Performance validation: developer-services/job/performance/431/

Accuracy validation: developer-services/job/accuracy/608/
Performance validation: developer-services/job/performance/428/

Accuracy validation: developer-services/job/accuracy/605/
Performance validation: developer-services/job/performance/421/

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/common/network_helper.cpp [hunk 1]
```diff
@@ -951,14 +951,38 @@ NetworkHelper::InsertDequantizationResult NetworkHelper::moveDequantizationAfter
         parent = std::make_shared<DequantizationConvert>(parent, dequantization.convert->get_output_element_type(0));
         ngraph::copy_runtime_info({ newOperation, parent }, parent);
     }
+
     if (moveSubtract && (dequantization.subtract != nullptr)) {
         auto subtractConstant = dequantization.subtract->get_input_node_shared_ptr(1);
-        parent = std::make_shared<DequantizationSubtract>(parent, subtractConstant);
+        const element::Type parentPrecision = parent->get_output_element_type(0);
+        if (parentPrecision.bitwidth() < subtractConstant->output(0).get_element_type().bitwidth()) {
+            THROW_IE_LPT_EXCEPTION(*parent) <<
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/common/network_helper.cpp:959]: Why cannot we insert convert for this case as well?

File: inference-engine/src/low_precision_transformations/src/common/subgraph.cpp [hunk 2]
```diff
@@ -86,6 +90,28 @@ bool Subgraph::fillSubgraphForQuantization(
     return true;
 }
 
+bool Subgraph::atLeastOneIsIntermediate(const std::shared_ptr<ngraph::Node>& node) const {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/common/subgraph.cpp:93]: I bet we need detailed comment what this method is actually check.

Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/common/subgraph.cpp:93]: Issue #37334 was extended

File: inference-engine/src/low_precision_transformations/src/common/network_helper.cpp (hunk 1 above)
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/common/network_helper.cpp:959]: was added in issue #42932

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**